cylc/flow/task_job_mgr.py

```diff
@@ -1281,6 +1282,35 @@ def _prep_submit_task_job_impl(self, workflow, itask, rtconfig):
             job_d=job_d
         )
 
+    @staticmethod
+    def get_execution_time_limit(
+        config_execution_time_limit: Any
+    ) -> Union[None, float]:
+        """Get execution time limit from config and process it.
+
+        We are making no assumptions about what can be passed in,
+        but passing silently and returning None if the value is
+        not a type convertable to a float:
+
+        Examples:
+            >>> this = TaskJobManager.get_execution_time_limit
+            >>> this(None)
+            >>> this("54")
+            54.0
+            >>> this({})
+
+        N.b.
+            This function does not save you from a value error:
+            >>> from pytest import raises
+            >>> with raises(ValueError):
+            ...     this("🇳🇿")
+        """
+        execution_time_limit = None
+        if config_execution_time_limit is not None:
+            with suppress(TypeError):
+                execution_time_limit = float(config_execution_time_limit)
+        return execution_time_limit
```

Comment:
I think this does the same?

```suggestion
        with suppress(TypeError):
            return float(config_execution_time_limit)
        return None
```